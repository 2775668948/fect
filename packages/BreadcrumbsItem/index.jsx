import { ref, watchEffect } from 'vue'
import { createNameSpace, useProvider } from '../utils'
import Link from '../Link'
const [createComponent] = createNameSpace('BreadcrumbsItem')
import './breadcrumbsItem.less'
import Separator from './breadcrumbs.spearator'
const READONLY_BREADCRUMBS_KEY = 'breadcrumbsKey'

export default createComponent({
  props: {
    to: {
      type: [String, Object],
      default: () => {},
    },
    href: {
      type: String,
      default: '',
    },
  },
  setup(props, { slots }) {
    const hasLink = ref(false)

    watchEffect(() => {
      /**
       * use Boolean to  display ,magic implicit convert is no encourage
       */
      const isNull = Boolean(props?.to) || props.href !== ''
      hasLink.value = isNull
    })
    const { ctx } = useProvider(READONLY_BREADCRUMBS_KEY)

    const content = slots.default?.()

    const withoutLinkRender = () => {
      return (
        <span class="fect-braed_item">
          {content} <Separator>{ctx.separator}</Separator>
        </span>
      )
    }
    const linkRender = () => {
      return (
        <Link class="fect-braed_item withLink" to={props.to} href={props.href}>
          {content}
          <Separator>{ctx.separator}</Separator>
        </Link>
      )
    }

    return () => (
      <>{hasLink.value ? <>{linkRender()}</> : <>{withoutLinkRender()}</>}</>
    )
  },
})
