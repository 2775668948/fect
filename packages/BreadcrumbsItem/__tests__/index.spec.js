import { mount } from '@vue/test-utils'
import BreadcrumbsItem from '../index'

describe('BreadcrumbsItem', () => {
  it('should render as element', () => {
    const breadcrumbsItem = mount(<BreadcrumbsItem />)
    expect(() => breadcrumbsItem.unmount()).not.toThrow()
  })
  it('when component have not to or href,it should render as span', () => {
    const wrapper = mount(<BreadcrumbsItem />)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.fect-braed_item').html()).toBe(
      '<span class="fect-braed_item"><!----><div class="fect-bread_separator"><!----></div></span>',
    )
  })
})
