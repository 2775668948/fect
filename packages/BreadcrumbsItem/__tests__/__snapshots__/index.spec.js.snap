// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbsItem when component have not to or href,it should render as span 1`] = `<span class="fect-braed_item"><!----> <div class="fect-bread_separator"><!----></div></span>`;
